libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1991,7 +2018,7 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 			// The old code generator had a generic "else" case here
 			// without any specific code being generated,
 			// but it would still be better to have an exhaustive list.
-			solAssert(false, "");
+			solAssert(noopMemberAccess, "");
```

Comment:
How can this be reachable from the line that sets this bool? Here we are inside `if (actualType.category() == Type::Category::Contract)`

Reply:
`if (actualType.category() == Type::Category::Contract)` ends up just before `else if (EnumType const* enumType = dynamic_cast<EnumType const*>(&actualType))`, so this was triggered before this change. I changed it because it was failing there...
Maybe it is better also to not use `noopMemberAccess` and use actual condition with empty block instead?

Reply:
Why not add
```
else if (auto const* arrayType = dynamic_cast<ArrayType const*>(&actualType))
	solAssert(arrayType->isByteArray() && member == "concat", "");
else
 	solAssert(false, "");
```
?
``if (init; cond)`` **is** nice, but no need to push it :-).